### ISSUE TITLE
TpetraCrsMatrix test coverage

### ISSUE THREAD

**Issue by tjfulle** [trilinos/ForTrilinos#120]

**Comment by tjfulle** [comment]
@aprokop, this test is ready, minus one addition to `fortpetra.f90` that would be very useful.  That is, an overload of `TpetraCrsMatrix%insertGlobalValues` with the following signature:

```
subroutine insertGlobalValues(this, globalrow, cols, vals)
  class(TpetraCrsMatrix) :: self
  integer(C_LONG_LONG), intent(in) :: globalrow
  integer(C_LONG_LONG), dimension(:) :: cols
  real(C_DOUBLE), dimension(:) :: vals
  ...
end subroutine
```

This would call the same `swigc_` procedure as the version with `insertGlobalValues(this, globalrow, nument, vals, inds)`, but would infer `nument` by `nument = size(cols)`.  

And similarly for `sumIntoGlobalValues`, `replaceGlobalValues`.  

I implemented all of these methods before, you can see them [here](https://github.com/trilinos/ForTrilinos/pull/112/files)

**Comment by aprokop** [comment]
@tjfulle OK, I'll take a look. I would prefer to first merge the MultiVector test, though.

**Comment by tjfulle** [comment]
@aprokop said:

> I would prefer to first merge the MultiVector test, though

See #122

**Comment by tjfulle** [comment]
@aprokop , I just pushed `test_tpetra_crsmatrix.f90`.  It does not compile right now, but will when the `TODO`s are resolved.  It compiled with a previous version of `fortpetra.f90` that I hand modified to include a few different `TpetraCrsMatrix` function overloads.  Once the necessary modifications are done in swig, it will compile.  There are some tests that don't pass, indicating some issues in `fortpetra.f90`.  They are also documented with `TODO`s (sorry, should have marked them `FIXME`).  The commit can be seen at 1c12e475f82cab8ba541b2f55085cc8832515ad6

**Comment by tjfulle** [comment]
Also, @aprokop, if you are confused what the `TODO: Remove`, `TODO: Reinstate` tags mean - the tests from Tpetra from which I copied these tests dealt only with `*LocalValues` methods, but these methods were not available in `fortpetra`, so I used the `*GlobalValues` version.  I'd like to remove the `*GlobalValues` versions in favor of the `*LocalValues` to have parity with the Tpetra tests.

**Comment by aprokop** [comment]
Fixed by #262.
